### DISCUSSION TITLE
Nullpointer in AmqpCreditBasedSender

### DISCUSSION THREAD

**Discussion by dv0gt** [smallrye/smallrye-reactive-messaging#2448] [Q&A]
Hi there,

I am experiencing Nullpointer Exceptions in the AmqpCreditBasedSender. It happens nearly every day, but I could not find out why ..

- I am using Quarkus Platform version 3.5.1
- I am using Azure Service Bus as message broker

The exception happens here:

![image](https://github.com/smallrye/smallrye-reactive-messaging/assets/120626623/2a9a65e8-ac8f-42fc-b254-5d30c1a1a0af)

I first thought this might be because of missing credits, but there should be no Nullpointer I think. This should be the line of code:

![image](https://github.com/smallrye/smallrye-reactive-messaging/assets/120626623/a11b62a4-dcd1-4836-8a47-ce38e133c946)

Do you have any idea why this can be null?

Do we need to check for open requests like mentioned [here](https://github.com/smallrye/smallrye-reactive-messaging/issues/1662#issuecomment-1066597969)? Or is this probably another issue?

Thanks for your help!

**Comment by cescoffier** [comment]
We would need to check what is `null`.
I would be surprise to have the vert.x instance `null`. Maybe there is a race condition on the holder access. 

Can you debug and tell us which part is `null`?

**Reply by dv0gt** [reply]
I will try to build a reproducer. It feels like a race condition, so its quite hard to debug.

**Reply by dv0gt** [reply]
@cescoffier what's interesting maybe, that this happens shortly after some connections have been re-established.

I know that Azure Service Bus closes connections sometimes, maybe that is not handled properly in this case ..

![image](https://github.com/smallrye/smallrye-reactive-messaging/assets/120626623/25284a14-d8c0-49b1-8b2a-6a1b8021b917)

I going to try to build a reproducer nevertheless, this is just for information

**Reply by cescoffier** [reply]
hum, the holder might be `null`.

**Reply by dv0gt** [reply]
Thats my assumption as well. Sadly I cannot really reproduce this on my local machine. We currently check for remaining credits and hope that this will happen less often.
